File: 7-kyu/find-all-non-consecutive-numbers/swift/solution.swift
func allNonConsecutive (_ arr: [Int]) -> [(Int, Int)] {
    var ret: [(Int, Int)] = []
    guard arr.count > 1 else { return ret }
    var properVal = arr[0]
    for (idx, val) in arr.enumerated() {
        if val != properVal {
            ret.append((idx, val))
            properVal = val
        }
        properVal += 1
    }
    return ret
}

File: 7-kyu/find-all-non-consecutive-numbers/javascript/solution.js
function allNonConsecutive(arr) {
  const nonConsecutive = [];
  for (let i = 1; i < arr.length; i++) {
    if (arr[i] !== arr[i - 1] + 1) {
      const next = { i: i, n: arr[i] };
      nonConsecutive.push(next);
    }
  }
  return nonConsecutive;
}
